hack/verify-generated.sh

```diff
@@ -0,0 +1,32 @@
+#!/bin/bash
+
+set -e
+
+tempdir="$(mktemp -d)"
+
+SRC_DIR="$(git rev-parse --show-toplevel)"
+
+cp "${SRC_DIR}/pkg/apis/hive/v1alpha1/zz_generated.deepcopy.go" "${tempdir}"
+cp "${SRC_DIR}/pkg/operator/assets/bindata.go" "${tempdir}"
+
+cd "${SRC_DIR}"
+make generate
+
+if ! diff -Naup "./pkg/apis/hive/v1alpha1/zz_generated.deepcopy.go" "${tempdir}/zz_generated.deepcopy.go"; then
+	echo "Generated deepcopy file is different. Run 'make generate'"
+	exit 1
+fi
+if ! diff -Naup "./pkg/operator/assets/bindata.go" "${tempdir}/bindata.go"; then
+	echo "Generated bindata file is different. Run 'make generate'"
+	exit 1
+fi
+
+mkdir "${tempdir}/crds"
+cp ${SRC_DIR}/config/crds/* "${tempdir}/crds/"
```

Comment:
Would it make sense to move this up before any of the generations?

I'm know that today the things in the `crds` directory are only modified by `make manifests`, but what if in the future something in the `crds` directory gets modified by `make generate` ?

In other words, why not copy all of the source compares before making any changes?

Reply:
ack, makes sense

Reply:
thx @twiest, code updated